tests/testthat/test-infection-integration.R

```diff
@@ -531,3 +560,195 @@ test_that('update_severe_disease renders with no infections', {
+  parameters <- set_clinical_treatment(parameters = parameters,
+                                       drug = 2,
+                                       timesteps = 1,
+                                       coverages = round(runif(1, 0, 1/2),
+                                                         digits = 2))
+  parameters <- set_antimalarial_resistance(parameters = parameters,
+                                            drug = 1,
+                                            timesteps = 1,
+                                            artemisinin_resistance = 1,
+                                            partner_drug_resistance = 0,
+                                            slow_parasite_clearance_prob = 0,
+                                            early_treatment_failure_prob = 1,
+                                            late_clinical_failure_prob = 0,
+                                            late_parasitological_prob = 0,
+                                            reinfection_prob = 0)
+  parameters <- set_antimalarial_resistance(parameters = parameters,
+                                            drug = 2,
+                                            timesteps = 1,
+                                            artemisinin_resistance = 1,
+                                            partner_drug_resistance = 0,
+                                            slow_parasite_clearance_prob = 0,
+                                            early_treatment_failure_prob = 1,
+                                            late_clinical_failure_prob = 0,
+                                            late_parasitological_prob = 0,
+                                            reinfection_prob = 0)
+  
+  clinical_infections <- individual::Bitset$new(100)
+  clinical_infections$insert(sample.int(n = 100, size = round(runif(n = 1, min = 10, max = 100)), replace = FALSE))
+  timestep <- 5
+  events <- create_events(parameters)
+  variables <- create_variables(parameters = parameters)
+  renderer <- individual::Render$new(timesteps = 10)
+  
+  treated <- calculate_treated(variables = variables,
+                               clinical_infections = clinical_infections,
+                               parameters = parameters,
+                               timestep = timestep,
+                               renderer = renderer)
+  
+  expect_identical(object = treated$size(), expected = 0, info = "Error: non-zero length treated Bitset when clinical treatment coverage = 1,
+                 artemisinin resistance = 1, and early treatment probability = 1")
+  expect_identical(renderer$to_dataframe()[timestep,'n_ETF'], renderer$to_dataframe()[timestep,'n_treated'], info = "Error: Number of
+                 early treatment failures does not match number of treated individuals when artemisinin resistance proportion and
+                 and early treatment failure probability both equal 1") 
```

Comment:
```suggestion
```

Reply:
Updated the test desc to "Number of treatment failures matches number of individuals treated when artemisinin resistance proportion and early treatment failure probability both set to 1" and implemented the suggested change to the test.